lib/hpke/src/digest.cpp

```diff
@@ -102,9 +139,21 @@ Digest::hmac_for_hkdf_extract(const bytes& key, const bytes& data) const
   // https://doi.org/10.6028/NIST.SP.800-131Ar2
   static const auto fips_min_hmac_key_len = 14;
   auto key_size = static_cast<int>(key.size());
+#if defined(WITH_OPENSSL3)
+  if (EVP_default_properties_is_fips_enabled(nullptr) &&
+      key_size < fips_min_hmac_key_len) {
+    // Currently OpenSSL's EVP_PKEY_new_mac_key API cannot deal with an empty
+    // key, see: https://github.com/openssl/openssl/issues/13089 As such, we
+    // have to use EVP_MAC API to do the HMAC operations. I am not sure if it's
+    // possible to do the following (or the equivalent) with EVP_MAC APIs:
+    //   EVP_MD_CTX_set_flags(context, EVP_MD_CTX_FLAG_NON_FIPS_ALLOW)
```

Comment:
Since the flag `EVP_MD_CTX_FLAG_NON_FIPS_ALLOW` has no effect from OpenSSL 3.0, I don't think we need to wait for this bug to be fixed. We should just use `EVP_MAC` APIs.

As far as I understand, OpenSSL doesn't seem to validate the key size of a HMAC in FIPS mode; it will validate whether an algorithm is FIPS approved, for example it will complain if we specify MD5 algorithm in FIPS mode. From what I've experimented so far, even if with just FIPS provider loaded, OpenSSL 3 doesn't complain if the HMAC key size is less than 112 bits, an expected digest value is always returned. Therefore, I think there is no need to check for the key size violation here. If in the future, OpenSSL becomes stricter in FIPS mode, we will need to handle this issue then. One way to handle this is to ensure both FIPS and default providers are loaded so that the shorter key can be provided by the fallback default provider.